### PR TITLE
Formatting updates + added isort

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose
This is another formatting update, with the following changes:
- added `isort` to sort imports in a deterministic fashion (run `isort -c .` to check, and `isort .` to sort)
- removed shebang at the beginning of various files
- removed unused `if __name__ == "__main__":` blocks
- moved some shared constants to the `utils` file
	- INIFINITY (1e20)
	- EPS (renamed from eps, fetched from numpy)

I've left this as a draft PR because I'm not entirely sure we want to use `isort`. I really like the concept but I'm hesitant to enforce it. Thoughts? In any case, if we don't want some stuff in this PR I can just open a new one and cherry-pick the commits we want.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Testing
All existing tests pass

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: pyoptsparse/pyNLPQLP/pyNLPQLP.py [hunk 1]
```diff
@@ -14,9 +14,6 @@
 from ..pyOpt_optimizer import Optimizer
 from ..pyOpt_error import Error
 
-# constants
-eps = np.finfo(np.float64).eps
```
Comment (marcomangano) [inline pyoptsparse/pyNLPQLP/pyNLPQLP.py:30]: what is this?

**Review by marcomangano — COMMENTED** [review]
Lots of interesting things here, but maybe we want to have an offline conversation about what is worth and what is not - e.g. I like `iSort` pretty much, but do we want to put the effort to test/maintain this formatting?

File: pyoptsparse/pyNLPQLP/pyNLPQLP.py (hunk 1 above)
Comment (nwu63) [reply pyoptsparse/pyNLPQLP/pyNLPQLP.py:30]: This is [machine epsilon](https://en.wikipedia.org/wiki/Machine_epsilon). We use it to compare design variables during hot start, for example.

**Review by marcomangano — APPROVED** [review]

**Review by sseraj — COMMENTED** [review]
Does this fix the OptView linting issues?

**Comment by nwu63** [comment]
> Does this fix the OptView linting issues?

Nope, there are a bunch more and I just couldn't be bothered to fix them. I guess I could though if you really wanted.

**Comment by sseraj** [comment]
No, that's alright.

File: pyoptsparse/pyIPOPT/pyIPOPT.py [hunk 2]
```diff
@@ -1,38 +1,26 @@
...
 pyIPOPT - A python wrapper to the core IPOPT compiled module.
 """
-# =============================================================================
-# IPOPT Library
-# =============================================================================
-
+# Compiled module
 try:
-    from . import pyipoptcore
+    from . import pyipoptcore  # isort: skip
```
Comment (sseraj) [inline pyoptsparse/pyIPOPT/pyIPOPT.py:6]: Can you explain why these have to be skipped?

Comment (nwu63) [reply pyoptsparse/pyIPOPT/pyIPOPT.py:6]: So there are two scenarios here.
1. For what you showed here, these are not necessary. If I don't skip them, then `isort` will just add a header here:
   ```
   -    from . import snopt  # isort: skip
   +    # Local modules
   +    from . import snopt
   ```
   This is a stylistic thing, and in the future I would like to move these imports inside the `__init__` which will eliminate this issue. However, if you think these are unnecessary we can definitely remove them.
2. There are other imports which are buried deep in the file. For example [this line](https://github.com/mdolab/pyoptsparse/blob/update-format/pyoptsparse/pyOpt_history.py#L220) imports the pyOptSparse version, and adding the skip prevents `isort` from adding an import header there which would be unnecessary.

**Review by sseraj — APPROVED** [review]

**Pull request merged**